### PR TITLE
Update OTEL to 1.6.0

### PR DESCRIPTION
**Pull request by luneo7**
Updates OpenTelemetry lib to 1.6.0, change it to use the dedicated autoconfigure SPI package instead of importing the autoconfigure directly, with this change we can remove the GraalVM  Autoconfiguration SDK substitution.

Adds the new option to enable gzip compression of the OTLP exporter payload.

And adjusts the `OtlpExporterSubstitutions` so it maps to the code change introduced by the v1.6.0 version

### CODE REVIEW

File: extensions/opentelemetry/opentelemetry-exporter-otlp/runtime/src/main/java/io/quarkus/opentelemetry/exporter/otlp/runtime/OtlpRecorder.java [hunk 1]
```diff
@@ -31,6 +31,10 @@ public void installBatchSpanProcessorForOtlp(OtlpExporterConfig.OtlpExporterRunt
                     Map<String, String> headers = OpenTelemetryUtil.convertKeyValueListToMap(runtimeConfig.headers.get());
                     headers.forEach(otlpGrpcSpanExporterBuilder::addHeader);
                 }
+                if (runtimeConfig.enableCompression) {
+                    // Currently, the only supported compression method is "gzip".
+                    otlpGrpcSpanExporterBuilder.setCompression("gzip");
```
Comment (kenfinnigan) [inline extensions/opentelemetry/opentelemetry-exporter-otlp/runtime/src/main/java/io/quarkus/opentelemetry/exporter/otlp/runtime/OtlpRecorder.java:36]: Is there an expectation for this to have alternatives in the future or no?

File: extensions/opentelemetry/opentelemetry-exporter-otlp/runtime/src/main/java/io/quarkus/opentelemetry/exporter/otlp/runtime/graal/OtlpExporterSubstitutions.java [hunk 2]
```diff
@@ -1,100 +1,51 @@
...
-                    throw new IllegalStateException(
-                            "TLS certificate configuration only supported with Netty. "
-                                    + "If you need to configure a certificate, switch to grpc-netty or "
-                                    + "grpc-netty-shaded.");
-                }
-                // TODO(anuraaga): Support okhttp.
-            }
-
-            channel = managedChannelBuilder.build();
+    public static void setTrustedCertificatesPem(
```
Comment (kenfinnigan) [inline extensions/opentelemetry/opentelemetry-exporter-otlp/runtime/src/main/java/io/quarkus/opentelemetry/exporter/otlp/runtime/graal/OtlpExporterSubstitutions.java:24]: Nice approach to narrow down the native image replacement, thanks!

**Review by kenfinnigan — COMMENTED** [review]
Can you also see if https://github.com/quarkusio/quarkus/pull/19897/files#diff-fa2d65cc1adc9844a13aca09417882484f1d3b32482c0bb5f7230a5d1e3bc774L29 can be made `static` again? It was made non static because autoconfigure was doing lookups, but my hope with us depending on the SPI only that this wouldn't happen.

File: extensions/opentelemetry/opentelemetry-exporter-otlp/runtime/src/main/java/io/quarkus/opentelemetry/exporter/otlp/runtime/OtlpRecorder.java (hunk 1 above)
Comment (luneo7) [reply extensions/opentelemetry/opentelemetry-exporter-otlp/runtime/src/main/java/io/quarkus/opentelemetry/exporter/otlp/runtime/OtlpRecorder.java:36]: According to the spec, gzip is the only supported, but language SDKs can provide other compressions that should be documented. Don't think that we will have another compression supported soon. What we can do be able to accommodate this change is leave as is, and in the feature add a parameter `compressionType` or `compressionCodec`. Or we can add it now and leave `gzip` as the default, we can leave this as a string and not use an enum, in this case if a codec is added to the Java SDK we don't need to change anything, or we can enforce using an enum, but whenever a new codec is added we would have to also update Quarkus to add the new codec.

**Comment by luneo7** [comment]
> Can you also see if https://github.com/quarkusio/quarkus/pull/19897/files#diff-fa2d65cc1adc9844a13aca09417882484f1d3b32482c0bb5f7230a5d1e3bc774L29 can be made `static` again? It was made non static because autoconfigure was doing lookups, but my hope with us depending on the SPI only that this wouldn't happen.

Will check, I think that we can move it back to static, since we won't have the auto configure in the class path to trigger build time compilation of it again

File: extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/restclient/ClientTracingFilter.java [hunk 3]
```diff
@@ -26,7 +26,7 @@
 
 @Priority(Priorities.HEADER_DECORATOR)
 public class ClientTracingFilter implements ClientRequestFilter, ClientResponseFilter {
-    private final TextMapPropagator TEXT_MAP_PROPAGATOR = GlobalOpenTelemetry.getPropagators().getTextMapPropagator();
+    private static final TextMapPropagator TEXT_MAP_PROPAGATOR = GlobalOpenTelemetry.getPropagators().getTextMapPropagator();
```
Comment (luneo7) [inline extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/restclient/ClientTracingFilter.java:29]: @kenfinnigan adding back the `static`, removing the autoconfigure sdk and using only the spi package fixed graalvm =]

File: extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/restclient/ClientTracingFilter.java [hunk 4]
```diff
@@ -98,10 +98,5 @@ public void filter(ClientRequestContext requestContext, ClientResponseContext re
         }
     }
 
-    private static final TextMapSetter<MultivaluedMap<String, Object>> SETTER = new TextMapSetter<MultivaluedMap<String, Object>>() {
-        @Override
-        public void set(MultivaluedMap<String, Object> carrier, String key, String value) {
-            carrier.add(key, value);
-        }
-    };
+    private static final TextMapSetter<MultivaluedMap<String, Object>> SETTER = MultivaluedMap::add;
```
Comment (luneo7) [inline extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/restclient/ClientTracingFilter.java:101]: Converted the anonymous class to a lambda expression

File: extensions/opentelemetry/opentelemetry-exporter-otlp/runtime/src/main/java/io/quarkus/opentelemetry/exporter/otlp/runtime/OtlpRecorder.java (hunk 1 above)
Comment (luneo7) [reply extensions/opentelemetry/opentelemetry-exporter-otlp/runtime/src/main/java/io/quarkus/opentelemetry/exporter/otlp/runtime/OtlpRecorder.java]: Used `compression` as an option string and it points to the otel spec, so whenever there is a new option we will be able to support right away.

File: extensions/opentelemetry/opentelemetry-exporter-otlp/runtime/src/main/java/io/quarkus/opentelemetry/exporter/otlp/runtime/OtlpExporterConfig.java [hunk 5]
```diff
@@ -48,5 +48,15 @@
...
 
+        /**
+         * Compression method to be used by exporter to compress the payload.
+         * <p>
+         * See <a href=
+         * "https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter.md#opentelemetry-protocol-exporter">
+         * Configuration Options</a> for the supported compression methods.
+         */
+        @ConfigItem()
+        public Optional<String> compression;
```
Comment (luneo7) [inline extensions/opentelemetry/opentelemetry-exporter-otlp/runtime/src/main/java/io/quarkus/opentelemetry/exporter/otlp/runtime/OtlpExporterConfig.java:59]: Added compression as an optional string, and it points to the spec configuration options. The SDK already validates the compression option and it throws an exception telling which codec is supported.

File: extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/restclient/ClientTracingFilter.java (hunk 3 above)
Comment (kenfinnigan) [reply extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/restclient/ClientTracingFilter.java:29]: That's great! Thanks

File: extensions/opentelemetry/opentelemetry-exporter-otlp/runtime/src/main/java/io/quarkus/opentelemetry/exporter/otlp/runtime/OtlpExporterConfig.java (hunk 5 above)
Comment (kenfinnigan) [reply extensions/opentelemetry/opentelemetry-exporter-otlp/runtime/src/main/java/io/quarkus/opentelemetry/exporter/otlp/runtime/OtlpExporterConfig.java:59]: Fantastic!

File: extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/restclient/ClientTracingFilter.java (hunk 4 above)
Comment (kenfinnigan) [reply extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/restclient/ClientTracingFilter.java:101]: Any particular reason?

For me, it's a little harder to immediately understand what's happening

**Comment by kenfinnigan** [comment]
@geoand are these two failures currently occurring in CI? Seems odd for these two continuous testing situations to fail with changes that aren't used in the tests.

@luneo7 can you verify these two tests don't fail locally for you?

**Comment by geoand** [comment]
I think they are flaky tests, yes, but @gastaldi has been following that more closely

**Comment by gastaldi** [comment]
Yeah, something else broke these tests. I haven't seen the `smallrye-reactive-messaging-kafka` failure but I remember seeing `LambdaDevServicesContinuousTestingTestCase` failing in other PRs

File: extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/restclient/ClientTracingFilter.java (hunk 4 above)
Comment (luneo7) [reply extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/restclient/ClientTracingFilter.java]: K, reverted no issues

**Review by kenfinnigan — APPROVED** [review]

**Pull request merged**